From ad0b119989926850e13ded78205a22c6b88603bb Mon Sep 17 00:00:00 2001
From: Constantin Musca <constantin.musca@intel.com>
Date: Thu, 19 Feb 2015 16:20:23 +0200
Subject: [PATCH] inv_mpu6050: write_raw: don't power off if the chip is
 enabled

If the chip is already enabled and the inv_mpu6050_write_raw
function is called then inv_mpu6050_set_power_itg(st, false)
will be called and the sensors will be disabled.
So, return -EBUSY and don't power off in this case.

Change-Id: Ief131d92603d8dec97604be0af00fb8425e12167
Tracked-On: https://jira01.devtools.intel.com/browse/IRDA-3810
Signed-off-by: Constantin Musca <constantin.musca@intel.com>
---
 drivers/iio/imu/inv_mpu6050/inv_mpu_core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c b/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
index ab9dc3b..504d6d3 100644
--- a/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
+++ b/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
@@ -454,8 +454,8 @@ static int inv_mpu6050_write_raw(struct iio_dev *indio_dev,
 	/* we should only update scale when the chip is disabled, i.e.,
 		not running */
 	if (st->chip_config.enable) {
-		result = -EBUSY;
-		goto error_write_raw;
+		mutex_unlock(&indio_dev->mlock);
+		return -EBUSY;
 	}
 	result = inv_mpu6050_set_power_itg(st, true);
 	if (result)
-- 
1.9.1

